crates/biome_configuration/src/javascript/formatter.rs

```diff
@@ -21,8 +21,16 @@ pub struct JavascriptFormatter {
     pub quote_properties: QuoteProperties,
 
     /// Print trailing commas wherever possible in multi-line comma-separated syntactic structures. Defaults to "all".
-    #[partial(bpaf(long("trailing-comma"), argument("all|es5|none"), optional))]
-    pub trailing_comma: TrailingComma,
+    #[partial(bpaf(
+        long("trailing-commas"),
+        // TODO: Remove in 2.0.0
+        long("trailing-comma"),
```

Comment:
This is [a hidden alias supported by `bpaf`](https://docs.rs/bpaf/latest/bpaf/_documentation/_1_tutorials/_2_derive_api/_1_custom_names/index.html#customizing-flag-and-argument-names). But there seems no easy way to generate a warning message if a user uses the alias.

Reply:
I'd make a standalone function and use it with external:

```rust
fn trailing_commas() -> impl Parser<TrailingCommas> {
    let primary = long("trailing-commas").help("Print ...").argument::<TrailingCommas>("all|es5|none");
    let compat = long("trailing comma").argument::<TrailingCommas>("ARG")
        .map(|x| { eprintln!("you are using deprecated option --blah, don't do that"); x }).hide();
    construct!([primary, compat]).optional()
}
```

And use it with something like `bpaf(external(trailing_commas)`


Not sure why you have `optional` there though. It implies `Option`... Is `TrailingCommas` an alias for something?

Reply:
@pacak Thank you very much!! The `optional` is a mistake I think. I'm new to rust and still struggling with these things 🙈

Reply:
Well, original one also comes with `optional` so maybe that's expected.